src/Settings/Default.hs

```diff
@@ -215,6 +215,44 @@ defaultBuilderArgs = mconcat
     , makeBuilderArgs
     , tarBuilderArgs ]
 
+-- | Disable some warnings in packages we use
+-- | https://github.com/ghc/ghc/blob/master/mk/warnings.mk#L46
+disableWarningArgsStage0 :: Args
+disableWarningArgsStage0 = stage Stage0 ? builder Ghc ? mconcat
+    [ package transformers ? mconcat [ arg "-fno-warn-unused-matches", arg "-fno-warn-unused-imports" ]
+    , package terminfo ? mconcat [ arg "-fno-warn-unused-imports" ] ]
+
+-- XXX: How to express Stage1 and upper?
+-- XXX: What about the commented-out packages?
```

Comment:
These packages are currently not considered part of GHC, but are still left in the tree (because at some point they shipped with GHC I believe). They may be gone in future. Or may be added back into packages built by default, in which case we'll add them to Hadrian.

This is not related to the build flavour.